### ISSUE TITLE
Problem when mac and IPv4 are defined

### ISSUE THREAD

**Issue by mbierman** [vectronic/homebridge-ping-hosts#21]
I recently started seeing this and I didn't understand the problem. 

```
[8/19/2022, 10:15:22 AM] [@vectronic/homebridge-ping-hosts] Restarting Process...
[8/19/2022, 10:15:23 AM] [@vectronic/homebridge-ping-hosts] Launched child bridge with PID 31420
[8/19/2022, 10:15:23 AM] Registering platform '@vectronic/homebridge-ping-hosts.PingHosts'
[8/19/2022, 10:15:23 AM] [@vectronic/homebridge-ping-hosts] Loaded @vectronic/homebridge-ping-hosts v3.1.4 child bridge successfully
[8/19/2022, 10:15:23 AM] Loaded 0 cached accessories from cachedAccessories.0E56B10B8D61.
[8/19/2022, 10:15:23 AM] [@vectronic/homebridge-ping-hosts] [Brother Online] closed_on_success: true
[8/19/2022, 10:15:23 AM] [@vectronic/homebridge-ping-hosts] [Brother Online] startup_as_failed: true
[8/19/2022, 10:15:23 AM] [@vectronic/homebridge-ping-hosts] [Brother Online] success_state: CONTACT_DETECTED
[8/19/2022, 10:15:23 AM] [@vectronic/homebridge-ping-hosts] [Brother Online] failure_state: CONTACT_NOT_DETECTED
[8/19/2022, 10:15:23 AM] [@vectronic/homebridge-ping-hosts] [Abode Online] closed_on_success: true
[8/19/2022, 10:15:23 AM] [@vectronic/homebridge-ping-hosts] [Abode Online] startup_as_failed: true
[8/19/2022, 10:15:23 AM] [@vectronic/homebridge-ping-hosts] [Abode Online] success_state: CONTACT_DETECTED
[8/19/2022, 10:15:23 AM] [@vectronic/homebridge-ping-hosts] [Abode Online] failure_state: CONTACT_NOT_DETECTED
[8/19/2022, 10:15:23 AM] [@vectronic/homebridge-ping-hosts] [Cameras Online] closed_on_success: true
[8/19/2022, 10:15:23 AM] [@vectronic/homebridge-ping-hosts] [Cameras Online] startup_as_failed: true
[8/19/2022, 10:15:23 AM] [@vectronic/homebridge-ping-hosts] [Cameras Online] success_state: CONTACT_DETECTED
[8/19/2022, 10:15:23 AM] [@vectronic/homebridge-ping-hosts] [Cameras Online] failure_state: CONTACT_NOT_DETECTED
[8/19/2022, 10:15:23 AM] [@vectronic/homebridge-ping-hosts] [Neurio Online] closed_on_success: true
[8/19/2022, 10:15:23 AM] [@vectronic/homebridge-ping-hosts] [Neurio Online] startup_as_failed: true
[8/19/2022, 10:15:23 AM] [@vectronic/homebridge-ping-hosts] [Neurio Online] success_state: CONTACT_DETECTED
[8/19/2022, 10:15:23 AM] [@vectronic/homebridge-ping-hosts] [Neurio Online] failure_state: CONTACT_NOT_DETECTED
[8/19/2022, 10:15:23 AM] [@vectronic/homebridge-ping-hosts] [BigMac Online] closed_on_success: true
[8/19/2022, 10:15:23 AM] [@vectronic/homebridge-ping-hosts] [BigMac Online] startup_as_failed: true
[8/19/2022, 10:15:23 AM] [@vectronic/homebridge-ping-hosts] [BigMac Online] success_state: CONTACT_DETECTED
[8/19/2022, 10:15:23 AM] [@vectronic/homebridge-ping-hosts] [BigMac Online] failure_state: CONTACT_NOT_DETECTED

/homebridge/node_modules/@vectronic/homebridge-ping-hosts/index.js:68
        self.log.error("[" + self.name + "] multiple addresses specified, ipv4_address will be used");
        ^
ReferenceError: self is not defined
    at new PingHostContactAccessory (/homebridge/node_modules/@vectronic/homebridge-ping-hosts/index.js:68:9)
    at PingHostsPlatform.accessories (/homebridge/node_modules/@vectronic/homebridge-ping-hosts/index.js:37:26)
    at /homebridge/node_modules/homebridge/src/bridgeService.ts:534:24
    at new Promise (<anonymous>)
    at BridgeService.loadPlatformAccessories (/homebridge/node_modules/homebridge/src/bridgeService.ts:528:12)
    at ChildBridgeFork.startBridge (/homebridge/node_modules/homebridge/src/childBridgeFork.ts:155:36)
[8/19/2022, 10:15:24 AM] [@vectronic/homebridge-ping-hosts] Child bridge process ended
[8/19/2022, 10:15:24 AM] [@vectronic/homebridge-ping-hosts] Process Ended. Code: 1, Signal: null
[8/19/2022, 10:15:31 AM] [@vectronic/homebridge-ping-hosts] Restarting Process...
[8/19/2022, 10:15:32 AM] [@vectronic/homebridge-ping-hosts] Launched child bridge with PID 31431
[8/19/2022, 10:15:32 AM] Registering platform '@vectronic/homebridge-ping-hosts.PingHosts'
[8/19/2022, 10:15:32 AM] [@vectronic/homebridge-ping-hosts] Loaded @vectronic/homebridge-ping-hosts v3.1.4 child bridge successfully
[8/19/2022, 10:15:32 AM] Loaded 0 cached accessories from cachedAccessories.0E56B10B8D61.
[8/19/2022, 10:15:32 AM] [@vectronic/homebridge-ping-hosts] [Brother Online] closed_on_success: true
[8/19/2022, 10:15:32 AM] [@vectronic/homebridge-ping-hosts] [Brother Online] startup_as_failed: true
[8/19/2022, 10:15:32 AM] [@vectronic/homebridge-ping-hosts] [Brother Online] success_state: CONTACT_DETECTED
[8/19/2022, 10:15:32 AM] [@vectronic/homebridge-ping-hosts] [Brother Online] failure_state: CONTACT_NOT_DETECTED
[8/19/2022, 10:15:32 AM] [@vectronic/homebridge-ping-hosts] [Abode Online] closed_on_success: true
[8/19/2022, 10:15:32 AM] [@vectronic/homebridge-ping-hosts] [Abode Online] startup_as_failed: true
[8/19/2022, 10:15:32 AM] [@vectronic/homebridge-ping-hosts] [Abode Online] success_state: CONTACT_DETECTED
[8/19/2022, 10:15:32 AM] [@vectronic/homebridge-ping-hosts] [Abode Online] failure_state: CONTACT_NOT_DETECTED
[8/19/2022, 10:15:32 AM] [@vectronic/homebridge-ping-hosts] [Cameras Online] closed_on_success: true
[8/19/2022, 10:15:32 AM] [@vectronic/homebridge-ping-hosts] [Cameras Online] startup_as_failed: true
[8/19/2022, 10:15:32 AM] [@vectronic/homebridge-ping-hosts] [Cameras Online] success_state: CONTACT_DETECTED
[8/19/2022, 10:15:32 AM] [@vectronic/homebridge-ping-hosts] [Cameras Online] failure_state: CONTACT_NOT_DETECTED
[8/19/2022, 10:15:32 AM] [@vectronic/homebridge-ping-hosts] [Neurio Online] closed_on_success: true
[8/19/2022, 10:15:32 AM] [@vectronic/homebridge-ping-hosts] [Neurio Online] startup_as_failed: true
[8/19/2022, 10:15:32 AM] [@vectronic/homebridge-ping-hosts] [Neurio Online] success_state: CONTACT_DETECTED
[8/19/2022, 10:15:32 AM] [@vectronic/homebridge-ping-hosts] [Neurio Online] failure_state: CONTACT_NOT_DETECTED
[8/19/2022, 10:15:32 AM] [@vectronic/homebridge-ping-hosts] [BigMac Online] closed_on_success: true
[8/19/2022, 10:15:32 AM] [@vectronic/homebridge-ping-hosts] [BigMac Online] startup_as_failed: true
[8/19/2022, 10:15:32 AM] [@vectronic/homebridge-ping-hosts] [BigMac Online] success_state: CONTACT_DETECTED
[8/19/2022, 10:15:32 AM] [@vectronic/homebridge-ping-hosts] [BigMac Online] failure_state: CONTACT_NOT_DETECTED

/homebridge/node_modules/@vectronic/homebridge-ping-hosts/index.js:68
        self.log.error("[" + self.name + "] multiple addresses specified, ipv4_address will be used");
        ^
ReferenceError: self is not defined
    at new PingHostContactAccessory (/homebridge/node_modules/@vectronic/homebridge-ping-hosts/index.js:68:9)
    at PingHostsPlatform.accessories (/homebridge/node_modules/@vectronic/homebridge-ping-hosts/index.js:37:26)
    at /homebridge/node_modules/homebridge/src/bridgeService.ts:534:24
    at new Promise (<anonymous>)
    at BridgeService.loadPlatformAccessories (/homebridge/node_modules/homebridge/src/bridgeService.ts:528:12)
    at ChildBridgeFork.startBridge (/homebridge/node_modules/homebridge/src/childBridgeFork.ts:155:36)
[8/19/2022, 10:15:33 AM] [@vectronic/homebridge-ping-hosts] Child bridge process ended
[8/19/2022, 10:15:33 AM] [@vectronic/homebridge-ping-hosts] Process Ended. Code: 1, Signal: null
[8/19/2022, 10:15:38 AM] [Flo notification] getStatus() http request returned http error code: 401
[8/19/2022, 10:15:38 AM] [Flo notification] Error occurred while pulling update from switch: Got html error code 401
[8/19/2022, 10:15:40 AM] [@vectronic/homebridge-ping-hosts] Restarting Process...
```

I then realized I had included both a mac and IPv4 address for a device in the configuration.
1. there should have been a warning in the UI. (nice to have)
2. the precedence should have been respected and the mac_address ignored. (personally I think the IP address should be ignored and the mac address used, but I suppose that's debatable). 

**Expected behavior**
Things should have worked regardless.

**Comment by mbierman** [comment]
Also noticed that with a valid and correct MAC address, one device was not discoverable for some reason. 

```
[8/19/2022, 10:35:21 AM] [@vectronic/homebridge-ping-hosts] [Obihai VOIP Online] response error: Error: Invalid IP address 'null' for null at Fri Aug 19 2022 10:35:21 GMT-0700 (Pacific Daylight Time) with session 739
```

**Comment by vectronic** [comment]
I will try to have a look at the issues you have raised in the coming week. Thanks for all of your feedback.

**Comment by vectronic** [comment]
This should be fixed in the latest version 3.2.0.
